lock.c

```diff
@@ -101,3 +109,55 @@ void iio_cond_signal(struct iio_cond *cond)
 	pthread_cond_signal(&cond->cond);
 #endif
 }
+
+#ifndef NO_THREADS
+static void * iio_thrd_wrapper(void *d)
+{
+	struct iio_thrd *thrd = d;
+
+	return (void *)(intptr_t) thrd->func(thrd->d);
```

Comment:
Why not keeping pthreads signature for the callback? It looks like we are not checking the return code anyways...

Reply:
Or maybe just use `pthread_exit()` and pass the return value to `pthread_join()`...

Reply:
This is needed because the prototype of the callbacks is different on Windows and Unix.

Reply:
Hmm I see. We could get away with some defines but I guess no need for that. Anyways, I would still consider have `pthread_exit()`... But no strong opinion, I'm also ok in leaving that decision to the real implementation of the thread (probably it's even safer)

Reply:
I wouldn't really use `pthread_exit()`, for instance I don't use in IIOD either. Since exiting the thread function does the exact same thing...

Reply:
I think the usage of `pthread_exit()` is mainly so you can pass a ret value to `pthread_join()`. But yeah, not that important and honestly I almost never remember to use either

Reply:
If you "return ptr;" in the thread function, you will retrieve "ptr" in `pthread_join()`, so that's no different.

Reply:
Hmm yes :facepalm: , it's implicit when you return from the thread function..